[BUG] ConsulSyncDataServiceTest.testWatch  AssertionError: expected:<1> but was:<0>

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Test cases fail to execute in some environments.
![image](https://user-images.githubusercontent.com/7974845/147812899-b7fd137d-a611-486c-bec9-0409f2c88c11.png)
there is some references to errors:
https://github.com/apache/incubator-shenyu/runs/4665747949?check_suite_focus=true

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

```markdown
ShenYu version(s):v2.4.1
```


### Debug logs

10:15:13.576 [shenyu-consul-config-watch-2] WARN org.apache.shenyu.sync.data.consul.ConsulSyncDataService - Error querying consul Key/Values for context 'shenyu/sync'. Message: null
10:15:13.678 [shenyu-consul-config-watch-3] WARN org.apache.shenyu.sync.data.consul.ConsulSyncDataService - Error querying consul Key/Values for context 'shenyu/sync'. Message: null
10:15:13.695 [main] WARN org.apache.shenyu.sync.data.consul.ConsulSyncDataService - Error querying consul Key/Values for context 'shenyu/sync'. Message: null
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.262 s <<< FAILURE! - in org.apache.shenyu.sync.data.consul.ConsulSyncDataServiceTest
Error:  org.apache.shenyu.sync.data.consul.ConsulSyncDataServiceTest.testWatch  Time elapsed: 0.367 s  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at org.apache.shenyu.sync.data.consul.ConsulSyncDataServiceTest.testWatch(ConsulSyncDataServiceTest.java:90)

[INFO] Running org.apache.shenyu.sync.data.consul.handler.ConsulCacheHandlerTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.212 s - in org.apache.shenyu.sync.data.consul.handler.ConsulCacheHandlerTest
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    ConsulSyncDataServiceTest.testWatch:90 expected:<1> but was:<0>
[INFO] 
Error:  Tests run: 8, Failures: 1, Errors: 0, Skipped: 0

### Anything else?

_No response_

--- a/shenyu-sync-data-center/shenyu-sync-data-consul/src/test/java/org/apache/shenyu/sync/data/consul/ConsulSyncDataServiceTest.java
+++ b/shenyu-sync-data-center/shenyu-sync-data-consul/src/test/java/org/apache/shenyu/sync/data/consul/ConsulSyncDataServiceTest.java
@@ -18,106 +18,84 @@
 package org.apache.shenyu.sync.data.consul;
 
 import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.QueryParams;
 import com.ecwid.consul.v1.Response;
 import com.ecwid.consul.v1.kv.model.GetValue;
 import org.apache.shenyu.common.constant.ConsulConstants;
-import org.apache.shenyu.common.utils.ReflectUtils;
 import org.apache.shenyu.sync.data.consul.config.ConsulConfig;
 import org.junit.After;
-import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
 
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.Objects;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.TimeUnit;
 
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.spy;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 /**
  * test case for {@link ConsulSyncDataService}.
  */
+@RunWith(MockitoJUnitRunner.Silent.class)
 public final class ConsulSyncDataServiceTest {
 
+    private static final int WAIT_TIME = 1000;
+
+    private static final int WATCH_DELAY = 1000;
+
+    private static final long INDEX = 1L;
+
+    @Mock
     private ConsulClient consulClient;
+
+    @Mock
+    private ConsulConfig consulConfig;
 
     private ConsulSyncDataService consulSyncDataService;
 
     @Mock
-    private Response<List<GetValue>> response;
+    private GetValue getValue;
 
     @Mock
-    private Map<String, Long> consulIndexes;
-
-    private AutoCloseable autoCloseable;
+    private Response response;
 
     @Before
     public void setup() {
-        consulClient = spy(ConsulClient.class);
-        ConsulConfig consulConfig = new ConsulConfig();
-        consulConfig.setWaitTime(1000);
-        consulConfig.setWatchDelay(1000);
+        when(consulConfig.getWaitTime()).thenReturn(WAIT_TIME);
+        when(consulConfig.getWatchDelay()).thenReturn(WATCH_DELAY);
+        final List<GetValue> list = new ArrayList<>(1);
+        when(getValue.getModifyIndex()).thenReturn(INDEX);
+        when(getValue.getKey()).thenReturn(ConsulConstants.PLUGIN_DATA);
+        when(getValue.getDecodedValue()).thenReturn("{}");
+        list.add(getValue);
+        when(consulClient.getKVValues(any(), any(), any()))
+                .thenReturn(response);
+        when(response.getValue()).thenReturn(list);
+        when(response.getConsulIndex()).thenReturn(INDEX);
         consulSyncDataService = new ConsulSyncDataService(consulClient, consulConfig, null,
                 Collections.emptyList(), Collections.emptyList());
-        AtomicBoolean running = (AtomicBoolean) ReflectUtils.getFieldValue(consulSyncDataService, "running");
-        assertTrue(Objects.requireNonNull(running).get());
-        autoCloseable = MockitoAnnotations.openMocks(this);
-    }
-
-    @After
-    public void after() throws Exception {
-        consulSyncDataService.close();
-        AtomicBoolean running = (AtomicBoolean) ReflectUtils.getFieldValue(consulSyncDataService, "running");
-        assertFalse(Objects.requireNonNull(running).get());
-        autoCloseable.close();
     }
 
     @Test
-    public void testWatch() throws InvocationTargetException, IllegalAccessException, NoSuchMethodException {
-        Long index = 1L;
-        final List<GetValue> list = new ArrayList<>();
-        GetValue getValue = spy(GetValue.class);
-        doReturn(index).when(getValue).getModifyIndex();
-        doReturn(ConsulConstants.PLUGIN_DATA).when(getValue).getKey();
-        doReturn("{}").when(getValue).getDecodedValue();
-        list.add(getValue);
-
-        doReturn(response).when(consulClient).getKVValues(any(), any(), any());
-        when(response.getValue()).thenReturn(list);
-        when(response.getConsulIndex()).thenReturn(1L);
-
-        Method watchConfigKeyValues = consulSyncDataService.getClass().getDeclaredMethod("watchConfigKeyValues");
-        watchConfigKeyValues.setAccessible(true);
-        watchConfigKeyValues.invoke(consulSyncDataService);
-
-        Object target = ReflectUtils.getFieldValue(consulSyncDataService, "consulIndexes");
-
-        consulIndexes = typeConversionMap(target);
-
-        Assert.assertEquals(index, consulIndexes.get(ConsulConstants.SYNC_PRE_FIX));
+    public void testStart() throws Exception {
+        TimeUnit.SECONDS.sleep(2);
+        verify(consulClient, atLeast(1)).getKVValues(eq(ConsulConstants.SYNC_PRE_FIX),
+                eq(null), any(QueryParams.class));
+        verify(getValue, atLeast(1)).getModifyIndex();
+        verify(getValue, atLeast(1)).getDecodedValue();
+        verify(response, atLeast(3)).getValue();
     }
 
-    private Map<String, Long> typeConversionMap(final Object obj) {
-        Map<String, Long> result = new HashMap<>();
-        if (obj instanceof Map<?, ?>) {
-            Map<?, ?> map = (Map<?, ?>) obj;
-            for (Map.Entry<?, ?> entry : map.entrySet()) {
-                result.put((String) entry.getKey(), (Long) entry.getValue());
-            }
-        }
-        return result;
+    @After
+    public void testStop() {
+        consulSyncDataService.close();
     }
 }

